$NetBSD: patch-Makefile,v 1.0 2024/08/12 23:00:00 dkazankov Exp $

Add gprbuild and gprinstall options
Make ALL_LIBRARY_TYPES externally settable

--- Makefile.orig	2024-01-03 19:00:52.000000000 +0200
+++ Makefile
@@ -4,19 +4,23 @@
 BUILD_DIR ?= .
 INSTALL_DIR ?= .
 
-ALL_LIBRARY_TYPES = static static-pic relocatable
+GPRBUILD_OPTIONS ?= -p -j$(PROCESSORS)
+GPRINSTALL_OPTIONS =
+
+ALL_LIBRARY_TYPES ?= static static-pic relocatable
 
 .PHONY: lib
 lib:
 	gprbuild -k -P adasat.gpr -p -j$(PROCESSORS) \
 		--relocate-build-tree="$(BUILD_DIR)" \
 		-XLIBRARY_TYPE=$(LIBRARY_TYPE) \
-		-XBUILD_MODE=$(BUILD_MODE)
+		-XBUILD_MODE=$(BUILD_MODE) \
+		$(GPRBUILD_OPTIONS)
 
 .PHONY: all-libs
 all-libs:
 	for kind in $(ALL_LIBRARY_TYPES) ; do \
-		gprbuild -k -P adasat.gpr -p -j$(PROCESSORS) \
+		gprbuild -k -P adasat.gpr $(GPRBUILD_OPTIONS) \
 			--relocate-build-tree="$(BUILD_DIR)" \
 			-XLIBRARY_TYPE=$$kind \
 			-XBUILD_MODE=$(BUILD_MODE) ; \
@@ -32,6 +36,7 @@
 			--prefix="$(INSTALL_DIR)" \
 			--build-name=$$kind \
 			--build-var=LIBRARY_TYPE ; \
+			$(GPRINSTALL_OPTIONS) \
 	done
 
 .PHONY: test
